packages/coding-standard/tests/Rules/CheckRequiredMethodTobeAutowireWithClassName/CheckRequiredMethodTobeAutowireWithClassNameTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Tests\Rules\CheckRequiredMethodTobeAutowireWithClassName;
+
+use Iterator;
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+use Symplify\CodingStandard\Rules\CheckRequiredMethodTobeAutowireWithClassName;
+
+final class CheckRequiredMethodTobeAutowireWithClassNameTest extends RuleTestCase
+{
+    /**
+     * @dataProvider provideData()
+     */
+    public function testRule(string $filePath, array $expectedErrorMessagesWithLines): void
+    {
+        $this->analyse([$filePath], $expectedErrorMessagesWithLines);
+    }
+
+    public function provideData(): Iterator
+    {
+        yield [__DIR__ . '/Fixture/EmptyDocblock.php', []];
+        yield [__DIR__ . '/Fixture/WithoutRequired.php', []];
+        yield [__DIR__ . '/Fixture/WithRequiredAutowire.php', []];
+        yield [
+            __DIR__ . '/Fixture/WithRequiredNotAutowire.php',
+            [[CheckRequiredMethodTobeAutowireWithClassName::ERROR_MESSAGE, 12]],
+        ];
+        yield [
+            __DIR__ . '/Fixture/WithRequiredPrivateAutowire.php',
+            [[CheckRequiredMethodTobeAutowireWithClassName::ERROR_MESSAGE, 12]],
+        ];
```

Comment:
I think this should be standalone rule, that requires method with the @require annotation to be public. 

And visibility skipped in this rule

Reply:
I've reverted public modifier check.